python/lsst/meas/extensions/gaap/gaap.py

```diff
@@ -0,0 +1,186 @@
+                       doc="List of sigmas for Gaussian apertures")
+    modelPsfMatch = ConfigurableField(doc="PSF Gaussianization Task", target=ModelPsfMatchTask)
+
+    def setDefaults(self) -> None:
+        # TODO: The following will move to a config file later in DM-27482
+        self.modelPsfMatch.kernel.active.alardNGauss = 1
+        self.modelPsfMatch.kernel.active.alardDegGaussDeconv = 1
+        self.modelPsfMatch.kernel.active.alardDegGauss = [8]
+        self.modelPsfMatch.kernel.active.alardGaussBeta = 1.0
+        self.modelPsfMatch.kernel.active.spatialKernelOrder = 0
+
+
+class BaseGaapFluxPlugin(GenericPlugin):
+
+    ConfigClass = BaseGaapFluxConfig
+
+    @classmethod
+    def getExecutionOrder(cls) -> float:
+        return cls.FLUX_ORDER
+
+    def getGaapResultName(self, sF: float, sigma: float) -> str:
+        return "_".join((self.name, str(sF).replace(".", "_")+"x", str(sigma).replace(".", "_")))
+
+    def __init__(self, config, name, schema, metadata) -> None:
+        """Gaussian Aperture and PSF (GAaP) photometry plugin.
+        """
+        GenericPlugin.__init__(self, config, name, schema, metadata)
+        for sF, sigma in itertools.product(self.config.scalingFactors, self.config.sigmas):
+            baseName = self.getGaapResultName(sF, sigma)
+            schema.addField(schema.join(baseName, "instFlux"), type="F", doc="GAaP Flux")
+            schema.addField(schema.join(baseName, "instFluxErr"), type="F", doc="GAaP Flux error")
+
+    def convolve(self, exposure: afwImage.Exposure, modelPsf: afwDetection.GaussianPsf,
+                 footprint: afwDetection.Footprint) -> afwImage.Exposure:
+        """Convolve the ``exposure`` to make the PSF same as ``modelPsf``.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Original (full) exposure containing all the sources.
+        modelPsf : `lsst.afw.detection.GaussianPsf`
+            Target PSF to which to match.
+        footprint : `lsst.afw.detection.Footprint`
+            Footprint of the source to measure.
+
+        Returns
+        -------
+        convExp : `lsst.afw.image.Exposure`
+            Sub-image containing the source, convolving to the target seeing
+        """
+
+        bbox = footprint.getBBox()
+        pixToGrow = self.config.modelPsfMatch.kernel.active.detectionConfig.fpGrowPix
+        # This is conservative for now.
+        # An alternate option is 2*max(self.psfMatch.kConfig.sizeCellX, self.psfMatch.kConfig.sizeCellY)
+        bbox.grow(pixToGrow)  # gets too big
+
+        origPsf = exposure.getPsf()
+
+        maskedImage = exposure.getMaskedImage()
+        subImage = maskedImage.Factory(maskedImage, bbox)
+        subExposure = afwImage.ExposureF(subImage)
+        subExposure.setPsf(origPsf)
+
+        config = self.config.modelPsfMatch
+        config.setDefaults()
+        config.kernel.active.alardSigGauss = [modelPsf.getSigma()]  # The size has to be set dynamically
```

Comment:
If I construct the task as I do now as opposed to constructing it in  `__init__` , am I also constructing the Task as many times as there are sources, as opposed to just once? And while it is ineffective, is there a measurable computational cost to setting up a new Task for each source?

Reply:
I still find your suggestion of passing the value of `alardSigGauss` while calling the `run` method, a bit hacky. The hacky part is just singling out just this one parameter. For instance, currently `alardNGauss` is fixed to be 1, but I could imagine a case where it needs to be set (in the future) to 2 or 3 depending on the PSF at the location of the source, which would also have to be set dynamically and hence passed to the `run` method. Would it make sense to generalize and say all non-optional config parameters, if set to `None` during initialization, can be supplied via the `run` method?

Reply:
> And while it is ineffective, is there a measurable computational cost to setting up a new Task for each source?

I doubt it is significant for a non-trivial measurement algorithm like this one.

> Would it make sense to generalize and say all non-optional config parameters, if set to `None` during initialization, can be supplied via the run method?

I think the fundamental issue here is that you are using a Task that was designed to be run once on entire images, not multiple times on different postage stamps.  So while it can be used for the latter, the fit isn't great.  I suspect the changes to the Task that would be necessary for it to be a natural fit are both bigger than generalizing _some_ config options into run parameters, but also much less than generalizing _all_ non-optional config options into parameters.

So if you can identify a few other config options that you think it's very likely you'd want to specialize per-object, it'd make sense to make them settable in `run` along with `alardSigGuess`, but I'd also try to keep that set small to avoid setting off a major refactor or adding complexity we'll never use.  "Adding complexity we'll never use" because it seems like it _might_ be useful is a super common trap to fall into - I think pretty much everybody who moves from working on a small codebase to a big one like ours does it a lot at first (certainly I did), because a codebase has to get to a certain size before the cost of just maintaining it starts to bite, but when it does, you really start to appreciate both simplicity and the philosophy of changing things only when needed.

Reply:
> I doubt it is significant for a non-trivial measurement algorithm like this one.

Sorry, do you mean that the cost of setting up a Task is most likely to be significant for a non-trivial algorithm?

>I think the fundamental issue here is that you are using a Task that was designed to be run once on entire images

I think the plugin framework is setup so that this needs to be done per-source, since all other sources would have been replaced with noise. In a way, this is better since you are not exacerbating the blending further by doing a further convolution in the presence of all sources.